### DISCUSSION TITLE
@solid-primitives/i18n

### DISCUSSION THREAD

**Discussion by arunbharath91** [solidjs-community/solid-primitives#549] [Q&A]
how to use it in solid start. iam getting error because dynamic imports.
if u r upgrading your package . plz provide example for both solid and solid start

**Comment by znycheporuk** [comment]
I've been struggling with the same question for some time, and here's what I came up with:
https://github.com/fpm-kpi-ua/website/blob/main/src/shared/use-translation.ts

There's a problem with invalidation — sometimes actions return the entire set of translations as a part of the response. I haven't investigated how to fix this yet; it likely has something to do with the invalidation keys, which I may have messed up.
